subsys/bluetooth/audio/ascs.c

```diff
@@ -507,6 +508,10 @@ static void ascs_cp_rsp_add(uint8_t id, uint8_t op, uint8_t code,
 	 */
 	case BT_ASCS_RSP_NOT_SUPPORTED:
 	case BT_ASCS_RSP_TRUNCATED:
+		if (rsp->num_ase) {
+			net_buf_simple_remove_mem(&rsp_buf,
+						  sizeof(*ase_rsp) * rsp->num_ase);
+		}
```

Comment:
And why should we do that here? In case of code 0x01 or 0x02 we are clearing all of the specif ASE's information from response, and what we should be left with is only opcode, num of ase which we set to 0xff, and then we add "current" id, code and reason - so in case of those errors we always should have a response with the length of five, of course in other cases (when everything goes well) we can check whether the length is correct but the check should look like this then:

Reply:
`	if (rsp->num_ase == 0xff) {
		__ASSERT_NO_MSG(rsp_buf.len == 5);
	} else {
		__ASSERT_NO_MSG(rsp_buf.len == (sizeof(*rsp) + sizeof(*ase_rsp) * rsp->num_ase));
	}
`

Reply:
We should never assert in the case of invalid packets from a peer, whenever that can be handled cleanly. Normally the right thing to do is to use BT_ WARN() and discard the packet. Asserts on the other hand should be reserved for catching local bugs in our code.

Reply:
This is a response that we are building, due to an error on our side - the peer did not send us any invalid packets in this case.